src/python_testing/TC_DRLK_2_13.py

```diff
@@ -0,0 +1,656 @@
+        for creds in all_aliro_cred_list:
+            await self.clear_credentials_cmd(credential=creds)
+
+    def generate_unique_octstr(self, length=65):
+        octal_digits = '01234567'
+        octstr = ''.join(random.choice(octal_digits) for _ in range(length))
+        return octstr.encode()
+
+    @async_test_body
+    async def test_TC_DRLK_2_13(self):
+        self.max_aliro_keys_supported = None
+        self.numberofcredentialsupportedperuser = None
+        self.signingKey = bytes.fromhex("89d085fc302ca53e279bfcdecdf3c4adb2f5d9bc9ea6c49e9566d144367df3ff")
+        self.verificationKey = bytes.fromhex(
+            "047a4c992d753924cdf3779a3c84fec2debaa6f0b3084450878acc7ddcce7856ae57b1ebbe2561015103dd7474c2a183675378ec55f1e465ac3436bf3dd5ca54d4")
+        self.groupIdentifier = bytes.fromhex("89d085fc302ca53e279bfcdecdf3c4ad")
+        self.groupResolvingKey = bytes.fromhex("89d0859bfcdecdf3c4adfc302ca53e27")
+        self.endpoint = 0
+        self.alirouser = "AliroUser"
+        self.alirocredentialissuerkey = bytes.fromhex(
+            "047a4c882d753924cdf3779a3c84fec2debaa6f0b3084450878acc7ddcce7856ae57b1ebbe2561015103dd7474c2a183675378ec55f1e465ac3436bf3dd5ca54d4")
+        self.alirocredentialissuerkey_secondary = bytes.fromhex(
+            "047a4c112d753924cdf3779a3c84fec2debaa6f0b3084450878acc7ddcce7856ae57b1ebbe2561015103dd7474c2a183675378ec55f1e465ac3436bf3dd5ca54d4")
+        self.aliroevictableendpointkey = bytes.fromhex(
+            "047a4c772d753924cdf3779a3c84fec2debaa6f0b3084450878acc7ddcce7856ae57b1ebbe2561015103dd7474c2a183675378ec55f1e465ac3436bf3dd5ca54d4")
+        self.alirononevictableendpointkey = bytes.fromhex(
+            "047a4c662d753924cdf3779a3c84fec2debaa6f0b3084450878acc7ddcce7856ae57b1ebbe2561015103dd7474c2a183675378ec55f1e465ac3436bf3dd5ca54d4")
+        self.alirononevictableendpointkey1 = bytes.fromhex(
+            "047a4c552d753924cdf3779a3c84fec2debaa6f0b3084450878acc7ddcce7856ae57b1ebbe2561015103dd7474c2a183675378ec55f1e465ac3436bf3dd5ca54d4")
+        #  step 1 TH reads DUT Endpoint 0 OperationalCredentials cluster CurrentFabricIndex attribute
+        self.step("1a")
+        self.fabric_idx1 = await self.read_attributes_from_dut(endpoint=self.endpoint,
+                                                               cluster=Clusters.Objects.OperationalCredentials,
+                                                               attribute=Clusters.OperationalCredentials.Attributes.CurrentFabricIndex
+                                                               )
+        if self.pics_guard(self.check_pics("DRLK.S.C1d.Rsp")):
+            self.step("1b")
+            await self.send_clear_user_cmd(user_index=int(0xFFFE))
+        if self.pics_guard(self.check_pics("DRLK.S.C26.Rsp")):
+            self.step("1c")
+            clear_all_credentials = Clusters.DoorLock.Structs.CredentialStruct(credentialIndex=int(0xFFFE))
```

Comment:
credentialType is not nullable - what will be the value? will it just be default (imagine 0, so it will be pin).
Test plan says to use null ... how does that work?

![image](https://github.com/user-attachments/assets/60de2923-698f-4492-8b3d-dd5bf7c9b7ef)

Reply:
![DRLKClearCred](https://github.com/user-attachments/assets/fa6ccd83-7a3b-4af0-977b-6ea42cc83b28)
The test plan uses null for credential type.

Reply:
code will now use null as per the updated spec

Reply:
I have updated the latest spec and re-built. The `CredentialType` is *not nullable*. If you are not setting it to any value, it will just take the default in python:

![image](https://github.com/user-attachments/assets/a163b050-b49c-424c-bf61-5771ca4a5222)

The test plan should be updated to say 0 or something else (and that made explicit in the test). You cannot set it to null because the value is not nullable.